framework/include/controls/ControllableParameter.h

```diff
@@ -14,141 +14,104 @@
+  /// Storage for the ControllableItems, these are stored as pointers to avoid copies.
+  std::vector<ControllableItem *> _items;
 };
 
-template <typename T>
-ControllableParameter<T>::ControllableParameter()
-{
-}
-
-template <typename T>
-void
-ControllableParameter<T>::insert(const MooseObjectParameterName & name,
-                                 std::shared_ptr<InputParameters> param_object)
-{
-  // Get a pointer to the Parameter
-  T * param_ptr = &param_object->template set<T>(name.parameter());
-
-  // Search for the pointer in the existing multimap
-  typename std::multimap<T *, MooseObjectParameterName>::iterator iter =
-      _parameter_names.find(param_ptr);
-
-  // If the pointer does not exist, add it
-  if (iter == _parameter_names.end())
-    _parameter_values.push_back(param_ptr);
-
-  // Update the object names
-  _parameter_names.insert(std::pair<T *, MooseObjectParameterName>(param_ptr, name));
-}
-
 template <typename T>
 void
-ControllableParameter<T>::set(const T & value)
-{
-  typename std::vector<T *>::iterator iter;
-  for (unsigned int i = 0; i < _parameter_values.size(); i++)
-    *_parameter_values[i] = value;
-}
-
-template <typename T>
-const std::vector<T *> &
-ControllableParameter<T>::get() const
+ControllableParameter::set(const T & value, bool type_check /*=true*/)
 {
-  return _parameter_values;
+  for (ControllableItem * item : _items)
+    item->set<T>(value, type_check);
 }
 
 template <typename T>
-std::string
-ControllableParameter<T>::dump()
+std::vector<T>
+ControllableParameter::get(bool type_check /*=true*/, bool warn_when_values_differ) const
 {
-  // Count of objects, for printing a number with the parameter
-  unsigned int index = 0;
-
-  // The output stream
-  std::ostringstream oss;
+  std::vector<T> output;
+  for (const ControllableItem * const item : _items)
+  {
+    std::vector<T> local = item->get<T>(type_check);
+    output.insert(output.end(), local.begin(), local.end());
+  }
 
-  // Loop through each pointer
-  typename std::vector<T *>::iterator iter;
-  for (iter = _parameter_values.begin(); iter != _parameter_values.end(); iter++)
+  // Produce a warning, if the flag is true, when multiple parameters have differing values
+  if (warn_when_values_differ && _items.size() > 1)
   {
-    // Get the list of names associated with the parameter
-    std::pair<typename std::multimap<T *, MooseObjectParameterName>::const_iterator,
-              typename std::multimap<T *, MooseObjectParameterName>::const_iterator>
-        range = _parameter_names.equal_range(*iter);
-    typename std::multimap<T *, MooseObjectParameterName>::const_iterator jt;
-
-    // Print each parameter as a list of possible names
-    oss << "(" << index << ") ";
-    for (jt = range.first; jt != range.second; ++jt)
-      oss << jt->second << " ";
-    oss << "\n";
-    index++;
+    // The first parameter to test against
+    const T value0 = output[0];
+
+    // Loop over all other parameter values
```

Comment:
For this comment to be true, the loop would have to start from index 1 which it does not.

But, I do not even see any reason to have this and the above comment in the code. The code is easy to read and understand.